play-json/jvm/src/main/scala/play/api/libs/json/jackson/JacksonJson.scala

```diff
@@ -257,7 +269,22 @@ private[jackson] class PlaySerializers(parserSettings: JsonParserSettings) exten
 }
 
 private[json] object JacksonJson {
-  private lazy val mapper = (new ObjectMapper).registerModule(new PlayJsonModule(JsonParserSettings.settings))
+  val defaultInstance: JacksonJson = JacksonJson(JsonConfig.settings)
+
+  private val ref: AtomicReference[JacksonJson] = new AtomicReference[JacksonJson](defaultInstance)
```

Comment:
So it add overhead for all runtime whereas it's only useful for testing?
Couldn't the internal set logic and its synchronization be only in tests?

Reply:
I agree if this is only for tests it probably doesn't need to be atomic. This is an internal API so we can control how it's used.

Also wondering if we could keep `JacksonJson` an object and just have a `setConfig` in there?

Reply:
Having the case class also means that any given operations will use a consistent set of mapper and factory, so there's not really a need for synchronization. Running tests concurrently could still give unexpected results (since it relies on static state), but the atomic reference didn't actually help with that anyway.